### PR TITLE
New and Improved Documented Simulator

### PR DESCRIPTION
**Pull request by akeley98**

### CODE REVIEW

File: src/lib/SimulatorLib/howto.txt [hunk 1]
```diff
@@ -0,0 +1,337 @@
+STSimulate.hs and the files in the SimulatorLib directory implement a
```
Comment (David-Durst) [inline src/lib/SimulatorLib/howto.txt]: Rename this file README.md and format it using markdown.

**Comment by akeley98** [comment]
Test comment

File: src/lib/STSimulate.hs [hunk 2]
```diff
@@ -5,25 +5,40 @@ import STMetrics
 import STAST
```
Comment (David-Durst) [inline src/lib/STSimulate.hs]: move this file to the simulate folder.

File: test/TestSimulator.hs [hunk 3]
```diff
@@ -71,7 +72,7 @@ simhlRandValues inRand seqLen T_Int =
   }
 simhlRandValues inRand seqLen T_Bit =
```
Comment (David-Durst) [inline test/TestSimulator.hs]: Remove the monad syntax.

File: test/TestSimulator.hs [hunk 4]
```diff
@@ -122,7 +123,7 @@ simhlMakeTestCases _ [] = []
 simhlMakeTestCases inRand (thisCase:cases) =
   do { let op = simhlTestOp thisCase
      ; let (memRand, inSeqs) = simhlOpRandInputs inRand op
```
Comment (David-Durst) [inline test/TestSimulator.hs]: Remove the monad syntax

**Review by David-Durst — CHANGES_REQUESTED** [review]

**Review by David-Durst — APPROVED** [review]
LGTM. Next time, please make this into multiple, smaller pull requests.

**Comment by David-Durst** [comment]
This pull request was undone by a hard push. I did not mean to accept it. I meant to accept https://github.com/David-Durst/aetherlingHaskellIR/pull/12 as this one had failing unit tests at the time of acceptance and didn't want to accept those into master.

**Pull request merged**